akd/src/append_only_zks.rs

```diff
@@ -242,25 +230,28 @@ impl Azks {
                     .await?;
             load_count += nodes.len() as u64;
 
-            current_nodes = Vec::<NodeKey>::new();
-
-            // Now that states are loaded in the cache, you can read and access them.
-            // Note, the two for loops are needed because otherwise, you'd be accessing remote storage
+            // Now that states are loaded in the cache, we can read and access them.
+            // Note, we perform directional loads to avoid accessing remote storage
             // individually for each node's state.
-            for node in &nodes {
-                if !nodes_to_load.contains(&node.label) {
-                    // Only continue to traverse nodes which are relevant prefixes to insertion_set
-                    continue;
-                }
-
-                for dir in crate::DIRECTIONS {
-                    let child_label = node.get_child_label(dir)?;
-                    if let Some(child_label) = child_label {
-                        current_nodes.push(NodeKey(child_label));
-                    }
-                }
-            }
+            current_nodes = nodes
+                .iter()
+                .filter(|node| filter_predicate(node))
+                .flat_map(|node| {
+                    DIRECTIONS
+                        .iter()
+                        .filter_map(|dir| {
+                            node.get_child_label(*dir)
+                                .expect(&format!(
```

Comment:
Oh oops, can we instead throw an error here instead of panicking if we encounter this?

Reply:
I actually think we may want to panic here. Reason being that the `DIRECTIONS` const *should only ever contain valid* directions: https://github.com/facebook/akd/blob/main/akd_core/src/types/mod.rs#L36-L37. I opted for a panic, as if we attempt to move in an invalid direction here, that seems like it would mean we may have a bug in lib.

Let me know what you think!

Reply:
See other comment about the function potentially throwing a StorageError. That said, in general, if we did have a situation where something should never happen, it would be better to see if we can make it enforced at compile-time rather than panicking at runtime.